app/code/Magento/CatalogInventory/Observer/AddStockItemsProductsObserver.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\CatalogInventory\Observer;
+
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
+use Magento\CatalogInventory\Api\StockConfigurationInterface;
+use Magento\CatalogInventory\Model\StockRegistryPreloader;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+/**
+ * Add Stock items to product collection.
```

Comment:
Please add deprecation info like in \Magento\CatalogInventory\Model\Plugin\AfterProductLoad

Reply:
I added deprecate comment